node-registrar/pkg/db/db.go

```diff
@@ -0,0 +1,59 @@
+package db
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+type Config struct {
+	Host        string
+	Port        int
+	DBName      string
+	User        string
+	Password    string
+	SSLMode     string
+	SqlLogLevel logger.LogLevel
+	MaxConns    int
+}
+
+// PostgresDatabase postgres db client
+type DataBase struct {
+	gormDB     *gorm.DB
+	connString string
+}
+
+var ErrRecordNotFound = errors.New("could not find any records")
+
+func NewDB(c Config) (db DataBase, err error) {
```

Comment:
I would try to split the concerns of the function, and ensure I do a ping on the DB when done